projects/step-frontend/src/lib/modules/timeseries/modules/_common/types/filter/filter-utils.ts

```diff
@@ -226,4 +226,45 @@ export class FilterUtils {
         throw new Error('Unrecognized type: ' + type);
     }
   }
+
+  public static createEmptyFilterTextItem(attribute: string): FilterBarItem {
+    const fieldType = this.getFilterFieldType(attribute);
+    return {
+      removable: true,
+      label: this.getFilterFieldLabel(attribute),
+      type: fieldType,
+      attributeName: attribute,
+      isLocked: false,
+    } as FilterBarItem;
+  }
+
+  private static getFilterFieldType(attribute: string): FilterBarItemType {
+    switch (attribute) {
+      case 'eId':
+        return FilterBarItemType.EXECUTION;
+      case 'taskId':
+        return FilterBarItemType.TASK;
+      case 'planId':
+        return FilterBarItemType.PLAN;
+      default:
+        return FilterBarItemType.FREE_TEXT;
+    }
+  }
+
+  public static isEntityFilter(filterItem: FilterBarItem): boolean {
+    return [FilterBarItemType.PLAN, FilterBarItemType.TASK, FilterBarItemType.EXECUTION].includes(filterItem.type);
+  }
+
+  private static getFilterFieldLabel(attribute: string): string {
+    switch (attribute) {
+      case 'eId':
+        return 'Execution';
+      case 'taskId':
+        return 'Task';
+      case 'plan':
+        return 'Plan';
+      default:
+        return attribute;
+    }
+  }
```

Comment:
I would prefer this to use the enum FilterBarItemType as a basis

Reply:
@neogucky I liked the idea, but I can't do it. This is called while iterating fields. some can be 'EXECUTION | PLAN, etc, but some other are FREE_TEXT - I still need the attribute as the label'. If I would change it to be based on FilterBarItemType, it would be `getFilterFieldLabel(type: FilterBarItemType, label: string)`. If you like this better, let me know, otherwise I'd keep it the way it is